### PR TITLE
Refactor spellcasting header and implementation (task #5339)

### PR DESCRIPTION
**Pull request by Capostrophic**
[Task 5339](https://gitlab.com/OpenMW/openmw/-/issues/5339)

They're now solely responsible for CastSpell class (which actually does spellcasting, what a surprise) instead of doing just about everything magic-related.

Resistance-related functions were moved to a new spellresistance header and various utility functions were moved to a new spellutil header.
Summoning-related utility functions were moved to summoning header because that makes sense. ApplyLoopingParticles visitor has been moved to the only place where it's used — to world implementation.
effectTick and some related functions were moved to a new tickableeffects header.
Absorb stat and reflect functionality has been moved to a new linkedeffects header.
Spell absorption functionality has been moved to a new spellabsorption header.
Some CastSpell class cleanup took place.

spellcasting.cpp file is now down ~650 lines compared to itself from 0.46.0. Some dependency clean up took place as well, so hopefully files that were previously dependent on spellcasting.hpp or are still dependent on it and spellcasting.cpp itself should recompile a bit faster in the future.

Once again it's not something obvious to the end user and I'd rather not add a changelog entry.

### CODE REVIEW

File: apps/openmw/mwmechanics/tickableeffects.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+namespace MWMechanics
+{
+    void adjustDynamicStat(CreatureStats& creatureStats, int index, float magnitude, bool allowDecreaseBelowZero = false)
+    {
+        DynamicStat<float> stat = creatureStats.getDynamic(index);
+        stat.setCurrent(stat.getCurrent() + magnitude, allowDecreaseBelowZero);
+        creatureStats.setDynamic(index, stat);
+    }
+
+    bool disintegrateSlot (const MWWorld::Ptr& ptr, int slot, float disintegrate)
```
[the remark this answers is not in the document]
Comment (Capostrophic) [reply apps/openmw/mwmechanics/tickableeffects.cpp:27]: I think it's doing fine here.

**Comment by Assumeru** [comment]
While you're at it anyway, maybe consider moving spell cost calculation to components so it can be reused by the CS for #2887.

**Comment by Capostrophic** [comment]
It can't be put to components as is, it needs esm store to work.

**Comment by Assumeru** [comment]
If you can template it into working shape that'd be cool. Otherwise, well, it's not like that issue hasn't been open forever anyway. It can wait a bit longer.

(Well it's not an app 🤷‍♀️)

**Comment by psi29a** [comment]
I'm taking it as-is. The work is done and the this isn't a high-traffic area anyway. Thank you @Capostrophic and @akortunov

**Pull request merged**